application/common/components/clients/GoogleSheetsClient.php

```diff
@@ -0,0 +1,229 @@
+        return $allUsersInfo;
+    }
+    
+    /**
+     * @param int $startRow
+     * @param int $howMany
+     * @return array
+     */
+    protected function getUsersInfoFromSpreadsheet(
+        int $startRow = 2,
+        int $howMany = 100
+    ) {
+        if (! $this->sheets instanceof \Google_Service_Sheets) {
+            $this->initGoogleClient();
+        }
+        
+        $users = [];
+        $range = sprintf('Users!A%s:L%s', $startRow, $startRow + $howMany - 1);
+        $rows = $this->sheets->spreadsheets_values->get(
+            $this->spreadsheetId,
+            $range,
+            ['majorDimension' => 'ROWS']
+        );
+        
+        if (isset($rows['values'])) {
+            foreach ($rows['values'] as $user) {
+                // If the first column is empty, take it to mean that there are no more records.
+                if (empty($user[0])) {
+                    break;
+                }
+                
+                // NOTE:
+                // Trailing empty cells are not returned by Google Sheets.
+                // Intermediate empty cells come back as empty strings, so an
+                // empty column could be absent or "". Handle both situations.
+                
+                $users[] = [
+                    User::EMPLOYEE_ID => $user[0],
+                    User::FIRST_NAME => $user[1],
+                    User::LAST_NAME => $user[2],
+                    User::DISPLAY_NAME => $user[3],
+                    User::USERNAME => $user[4],
+                    User::EMAIL => $user[5],
+                    User::ACTIVE => $user[6] ?? 'yes',
+                    User::LOCKED => $user[7] ?? 'no',
+                    'last_synced' => $this->getValueIfNonEmpty($user, 8),
+                    User::REQUIRE_MFA => $user[9] ?? 'no',
+                    User::MANAGER_EMAIL => $this->getValueIfNonEmpty($user, 10),
+                    User::SPOUSE_EMAIL => $this->getValueIfNonEmpty($user, 11),
+                ];
+            }
+        }
+        
+        return $users;
+    }
+    
+    /**
+     * Get the value in the array at the specified index (or key). If empty,
+     * return null.
+     *
+     * @param array $array The array to get the value from.
+     * @param int|string $index The index (or key) whose value is desired.
+     * @return mixed|null The resulting non-empty value, or null.
+     */
+    protected function getValueIfNonEmpty($array, $index)
+    {
+        if (empty($array[$index])) {
+            return null;
+        }
+        return $array[$index];
+    }
+    
+    public function updateSyncDatesFor(array $employeeIds)
+    {
+        $employeeIdsAsStrings = array_map(function($employeeId) {
+            return (string)$employeeId;
+        }, $employeeIds);
+        
+        if (! $this->sheets instanceof \Google_Service_Sheets) {
+            $this->initGoogleClient();
+        }
+        
+        $nowAsADateString = date('c');
+        
+        $startAtRow = 2;
```

Comment:
Low importance: If these are used multiple places, would it be better to have them as class level constants?